config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -32,59 +41,155 @@
         // treeset is needed here to sort the builders in the proper order to get a correct bundle builded
         // Ordering is necessary for the bundle, for the gateway to load it properly.
         this.entityBuilders = unmodifiableSet(new TreeSet<>(entityBuilders));
         this.bundleDocumentBuilder = bundleDocumentBuilder;
         this.bundleMetadataBuilder = bundleMetadataBuilder;
+        this.documentTools = documentTools;
     }
 
-    public Map<String, Pair<Element, BundleMetadata>> build(Bundle bundle, EntityBuilder.BundleType bundleType,
-                                                            Document document, String projectName,
-                                                            String projectGroupName, String projectVersion) {
+    public Pair<Element, BundleMetadata> build(Bundle bundle, EntityBuilder.BundleType bundleType,
+                                               Document document, String projectName,
+                                               String projectGroupName, String projectVersion, AnnotatedEntity annotatedEntity) {
         List<Entity> entities = new ArrayList<>();
         entityBuilders.forEach(builder -> entities.addAll(builder.build(bundle, bundleType, document)));
+        if (annotatedEntity != null) {
+            if (EntityBuilder.BundleType.DEPLOYMENT == bundleType) {
+                List<Entity> entityList = getEntityDependencies(annotatedEntity.getPolicyName(), entities, bundle);
+                LOGGER.log(Level.FINE, "Annotated entity list : " + entityList);
+                Optional<Entity> rootEntity = entityList.stream().filter(entity -> annotatedEntity.getEntityName().equals(entity.getName())
+                                                                                            && annotatedEntity.getEntityType().equals(entity.getType())).findFirst();
+                if (!rootEntity.isPresent()) {
+                    Optional<Entity> foundEntity = entities.stream().filter(entity -> annotatedEntity.getEntityName().equals(entity.getName())
+                                                                                            && annotatedEntity.getEntityType().equals(entity.getType())).findFirst();
+                    if (foundEntity.isPresent()) {
+                        entityList.add(foundEntity.get());
+                    }
+                }
+                List<Entity> bundleEntities = renameNonReusableEntities(entityList, bundle, annotatedEntity, projectGroupName, projectVersion);
+                // Create bundle and its metadata
+                final Element annotatedBundle = bundleDocumentBuilder.build(document, bundleEntities);
+                final BundleMetadata bundleMetadata = bundleMetadataBuilder.build(annotatedEntity,
+                        bundleEntities, projectGroupName, projectVersion);
+
+                return ImmutablePair.of(annotatedBundle,
+                        bundleMetadata);
```

Comment:
This can be moved in the same line. `ImmutablePair.of(annotatedBundle, bundleMetadata);`